src/apps/frontend/pages/authentication/phone-login/phone-login-form.hook.ts

```diff
@@ -0,0 +1,71 @@
+import { useFormik } from 'formik';
+import { PhoneNumberUtil } from 'google-libphonenumber';
+import { useNavigate } from 'react-router-dom';
+import * as Yup from 'yup';
+
+import constant from '../../../constants';
+import constants from '../../../constants/routes';
+import { useAuthContext } from '../../../contexts';
+import { AsyncError } from '../../../types';
+
+interface PhoneLoginFormProps {
+  onSuccess: () => void;
+  onError: (err: AsyncError) => void;
+}
+const usePhoneLoginForm = ({ onSuccess, onError }: PhoneLoginFormProps) => {
+  const {
+    isSendOTPLoading, sendOTPError, sendOTP,
+  } = useAuthContext();
+
+  const navigate = useNavigate();
+
+  const formik = useFormik({
+    initialValues: {
+      countryCode: '+1',
+      country: 'US',
+      phoneNumber: '',
+    },
+    validationSchema: Yup.object({
+      phoneNumber: Yup.string()
+        .min(8, constant.PHONE_VALIDATION_ERROR)
+        .max(14, constant.PHONE_VALIDATION_ERROR)
+        .required(constant.PHONE_VALIDATION_ERROR),
+    }),
+    onSubmit: (values) => {
+      const validateAndFormatPhoneNumber = (phoneNumber: string, countryCode: string) => {
```

Comment:
Let's keep this simple and step-by-step -
1) parsedPhoneNumber (parse the phone number)
2) isValidPhoneNumber check, if not valid show an error and return;
3) if a valid phone number then get formattedPhoneNumber
4) sendOtp